magma/tuple.py

```diff
@@ -188,28 +188,28 @@ def __eq__(cls, rhs):
 class Tuple(Type, Tuple_, metaclass=TupleKind):
     def __init__(self, *largs, **kwargs):
 
-        Type.__init__(self, **kwargs) # name=
+        Type.__init__(self, **kwargs)  # name=
 
-        self.ts = []
+        self.ts = {}
         if len(largs) > 0:
             assert len(largs) == len(self)
-            for k, t, T in zip(self.keys(), largs, self.types()):
+            for i, (k, t, T) in enumerate(zip(self.keys(),
+                                              largs,
+                                              self.types())):
                 if isinstance(t, (IntegerTypes, BitVector, Bit)):
                     t = T(t)
-                self.ts.append(t)
-                if not isinstance(self, AnonProduct):
-                    setattr(self, k, t)
-        else:
```

Comment:
Because this else case is responsible for populating `ts` in the default case (when it's not constructed with largs), but now we populate ts on demand